### ISSUE TITLE
code object is not signed at all

### ISSUE THREAD

**Issue by TomRomanowski** [electron-userland/electron-builder#3169]
* **Version**: 20.25.0

* **Target**: macos

When I build for windows using NSIS, I have no issue signing the app. However, when I try to build for macos, I get `code object is not signed at all` because of my LICENSE.txt being in the Contents directory. It's the same issue here:  https://github.com/electron-userland/electron-builder/issues/2522
but it doesn't seem like it was fixed

In my package.json, I have:
```
{
   ...
   "build": {
      ...
      "extraFiles": "LICENSE.txt"
   }
}
```
I created an afterPack script to move the file myself but the same issue persists

**Comment by pronebird** [comment]
We experience the same with additional binary (executable) added to `extraFiles`. Shouldn't electron-builder sign it?

**Comment by fishear** [comment]
I also have the same problem, does anybody have a resolution?

**Comment by fishear** [comment]
> We experience the same with additional binary (executable) added to `extraFiles`. Shouldn't electron-builder sign it?

I also encountered the same problem, could you please tell me do you have a workaround? thanks~~

**Comment by fishear** [comment]
> * **Version**: 20.25.0
> 
> * **Target**: macos
> 
> When I build for windows using NSIS, I have no issue signing the app. However, when I try to build for macos, I get `code object is not signed at all` because of my LICENSE.txt being in the Contents directory. It's the same issue here: #2522
> but it doesn't seem like it was fixed
> 
> In my package.json, I have:
> 
> ```
> {
>    ...
>    "build": {
>       ...
>       "extraFiles": "LICENSE.txt"
>    }
> }
> ```
> I created an afterPack script to move the file myself but the same issue persists

Hi, did you fix it? I tried my best, but still failed.

**Comment by X140Yu** [comment]
Same issue here, has anyone figured out how to deal with it?

**Comment by Christilut** [comment]
Same problem here

**Comment by Circuit8** [comment]
Same here, any solution?

**Comment by Christilut** [comment]
My solution was simple, don't use a file without extension. It seems to assume that without extension it is a binary that must be signed. I just changed my `LICENSE` file to `LICENSE.txt`

If you can't do that, I'd suggest signing it manually with the `codesign` CLI tool, it's really easy.

**Comment by Circuit8** [comment]
@Christilut Thanks a lot thats really helpful. My file was an exectuable, I thought electron-builder would handle signing it, but turns out youre right, I had to sign it first, then it gets signed again as part of the final .app file by electron-builder.

**Comment by GeorgeBarlow** [comment]
> @Christilut Thanks a lot thats really helpful. My file was an exectuable, I thought electron-builder would handle signing it, but turns out youre right, I had to sign it first, then it gets signed again as part of the final .app file by electron-builder.

I'm sorry to bother you, but could you give an example of how you codesigned it all, I'm really new to codesigning and I just wanted to know the path you took to rectify this, having the exact same issue.

**Comment by Circuit8** [comment]
@GeorgeBarlow 
It's been a while since I've touched it so this may be partially innacurate. But basically in my `package.json` I have:

```JSON
    "win": {
      "icon": "build/icons/icon.ico",
      "extraFiles": [
        {
          "from": "./prebuilt_binaries/windows/cjpeg.exe",
          "to": "binaries/cjpeg.exe"
        },
        {
          "from": "./prebuilt_binaries/windows/dcraw.exe",
          "to": "binaries/dcraw.exe"
        }
      ]
    },
```

Which essentially tells electron builder to copy dcraw and cjpeg into the distributed executable. From what I can remember these two files need to be signed using a windows CLI tool, presumably using the key you're going to sign your electron app with. They only need to be signed once, electron builder will include them every time it builds, then sign the whole executable and you're good to go.

Sorry I can't be more specific it's been a while since I've touched it! Code signing is a huge pain but you'll get there.
